templates/base/client/prototype-ui/State.tsx.hbs

```diff
@@ -158,18 +160,30 @@ function PluginDisplay<T>(props: { value: T; component: string; children: (value
       ref.current.updatedAt = updatedAt;
       (ref.current as any).addEventListener("error", displayError);
       return () => {
-        (ref.current as any).removeEventListener("error", displayError);
+        if (ref.current) {
+          (ref.current as any).removeEventListener("error", displayError);
+        }
       };
     }
   });
-  return <div className="p-1 plugin-display">{React.createElement(props.component, { ref })}</div>;
+            console.log(props.children);
+  if (_pluginsAsObjects) {
+    return (<>
+      {props.children(props.value)}
+          </>
+    );
+  }
+  else {
+    return <div className="p-1 plugin-display">{React.createElement(props.component, { ref })}</div>;
+  }
 }
 
 {{#each types}}
 {{> renderComponent}}
 {{/each}}
-export function State() {
+export function State({ pluginsAsObjects }: { pluginsAsObjects: boolean; }) {
```

Comment:
i think it might be cleaner to include this as part of the `HathoraContext`. then the `PluginDisplay` can access it directly via its `useContext`

Reply:
good call, fixed